app/src/main/java/org/schabi/newpipe/util/ShortcutsHelper.java

```diff
@@ -0,0 +1,185 @@
+package org.schabi.newpipe.util;
+
+import android.annotation.SuppressLint;
+import android.app.ActivityManager;
+import android.content.Context;
+import android.content.Intent;
+import android.content.pm.ShortcutInfo;
+import android.content.pm.ShortcutManager;
+import android.graphics.Bitmap;
+import android.graphics.BitmapFactory;
+import android.graphics.Canvas;
+import android.graphics.Paint;
+import android.graphics.PorterDuff;
+import android.graphics.PorterDuffXfermode;
+import android.graphics.Rect;
+import android.media.ThumbnailUtils;
+import android.os.Build;
+
+import androidx.annotation.DrawableRes;
+import androidx.annotation.NonNull;
+import androidx.annotation.Nullable;
+import androidx.core.content.pm.ShortcutInfoCompat;
+import androidx.core.content.pm.ShortcutManagerCompat;
+import androidx.core.graphics.drawable.IconCompat;
+
+import org.schabi.newpipe.BuildConfig;
+import org.schabi.newpipe.MainActivity;
+import org.schabi.newpipe.R;
+import org.schabi.newpipe.extractor.channel.ChannelInfoItem;
+
+import java.io.InputStream;
+import java.util.Collections;
+import java.util.List;
+
+import io.reactivex.Single;
+import io.reactivex.android.schedulers.AndroidSchedulers;
+import io.reactivex.schedulers.Schedulers;
+import okhttp3.OkHttpClient;
+import okhttp3.Request;
+import okhttp3.Response;
+
+public final class ShortcutsHelper {
+
+	public static final String ACTION_OPEN_SHORTCUT = "org.schabi.newpipe.action.OPEN_SHORTCUT";
+
+	private ShortcutsHelper() {
+	}
+
+	@SuppressLint("CheckResult")
+	@SuppressWarnings("ResultOfMethodCallIgnored")
```

Comment:
Is it ok to ignore Disposables? Couldn't this lead to crashes (when an activity is closed but its disposables are not disposed) or leaks?

Reply:
I think this is ok, because:
1. This operation is relatively short, so should not be any memory leaks
2. We want to complete this action even if activity was destroyed. Of course, this operation can be placed into separated Service, but, i think, it is too small